campaign-launcher/interfaceV2/src/hooks/recording-oracle/exchangeApiKeys.ts

```diff
@@ -1,16 +1,58 @@
-import { useQuery } from "@tanstack/react-query";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
 import { useAccount } from "wagmi";
 
 import { recordingApi } from "../../api";
+import { QUERY_KEYS } from "../../constants/queryKeys";
 import { useWeb3Auth } from "../../providers/Web3AuthProvider";
 
-export const useGetExchangeApiKeys = () => {
+export const useGetEnrolledExchanges = () => {
   const { isAuthenticated } = useWeb3Auth();
   const { isConnected } = useAccount();
 
   return useQuery({
-    queryKey: ['exchange-api-keys'],
-    queryFn: () => recordingApi.get('/exchange-api-keys'),
+    queryKey: [QUERY_KEYS.ENROLLED_EXCHANGES],
+    queryFn: () => recordingApi.getEnrolledExchanges(),
     enabled: isAuthenticated && isConnected,
   });
+}
+
+export const useGetExchangesWithApiKeys = () => {
+  const { isAuthenticated } = useWeb3Auth();
+  const { isConnected } = useAccount();
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.EXCHANGES_WITH_API_KEYS],
+    queryFn: () => recordingApi.getExchangesWithApiKeys(),
+    enabled: isAuthenticated && isConnected,
+  });
+};
+
+type MutationPayload = {
+  exchangeName: string;
+  apiKey: string;
+  secret: string;
+}
+
+export const usePostExchangeApiKey = () => {
```

Comment:
For this and `delete` mutations, should we also invalidate `ENROLLED_EXCHANGES` query, so list of exchange names is actual after that as well?

Reply:
@dnechay i thought about this, but it seems like we gonna use those queries on different pages, so no need to revalidate immediately